packages/vite/src/node/optimizer/index.ts

```diff
@@ -624,23 +581,66 @@ export function runOptimizeDeps(
           }
         }
 
-        const dataPath = path.join(processingCacheDir, '_metadata.json')
-        writeFile(
-          dataPath,
-          stringifyDepsOptimizerMetadata(metadata, depsCacheDir),
-        )
-
         debug(
           `Dependencies bundled in ${(performance.now() - start).toFixed(2)}ms`,
         )
 
-        return createProcessingResult()
+        return {
+          metadata,
+          async commit() {
+            // Write this run of pre-bundled dependencies to the deps cache
+
+            let oldFilesNames: string[]
```

Comment:
Not a big deal, but I was thinking keeping a list of new file names instead and then when we cleanup, we traverse the dir files and delete those that aren't new 🤔

Reply:
This is what `cleanupDepsCacheStaleFiles` does 🤔
The list of new files is already in `metadata` and we are checking the old files against the new ones. Maybe I'm missing something here.